### PR TITLE
No typing activity if bot is running as a skill

### LINKED ISSUE #2783: Avoid sending typing activity when bot is invoked as skill
We should port this once the C# PR is merged.
See [parent](https://github.com/microsoft/botframework-sdk/issues/6049)

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes #2783

## Description
Ports functionality from dotnet SDK to not send typing indicators if a bot is running as a skill.

## Specific Changes
- Adds a TestSkillAdapter helper class that automatically includes
claims to mimic a Skills bot
- Adds logic to ShowTypingMiddleware to skip indicators if bot is
running as a skill
- Adds a test for this behavior

## Testing
Added a test case.

### CODE REVIEW

**Review by gabog — APPROVED** [review]
The only observation here is that I would have created testSkillAdapter as a private class in the test case (unless we are planning to reuse it else where).

That said, I like this approach better than the original C# version and backported the concepts in this PR https://github.com/microsoft/botbuilder-dotnet/pull/4701

**Comment by joshgummersall** [comment]
@gabog I imagine we may extend our test suite with more "bot running as a skill" scenarios which was the main reason for exporting the class. I'm happy to internalize that class within the typing middleware test if you think that's the wrong approach.

**Comment by gabog** [comment]
not a big deal @joshgummersall. we can merge it as is, if we find ourselves using that helper class over and over, then let's think how we can factor this into the product.

**Comment by gabog** [comment]
not a big deal @joshgummersall. we can merge it as is, if we find ourselves using that helper class over and over, then let's think how we can factor this into the product.

**Comment by joshgummersall** [comment]
I'll internalize the class before merging just to reduce our surface area until we decide on it's reusability. I also have to address an issue with the `botframework-connector` as well so we can't merge yet anyways.

File: libraries/botbuilder-core/package.json
Comment (stevengum) [inline libraries/botbuilder-core/package.json:24]: We've avoided adding `botframework-connector` to `botbuilder-core` because of its various dependencies that run only on the Node.js runtime. `botbuilder-core`, `botbuilder-ai`, `botbuilder-dialogs` and other packages have been shipped with a "runnable-in-Node.js-or-browsers" stance; which means we can't add `botframework-connector` as a dependency.

Ideally the package/code in `botframework-connector` would have been factored or shipped (bundled/packed) in a way to allow for the browser-compatible packages to take a direct dependency on the necessary code.

There's a few paths we could take to safely consume the necessary `SkillValidation` code in `botbuilder-core`:

1. Factor out the code into a new package that's consumable by any environment
2. Duplicate the code
3. Add environment integration tests to make sure that botframework-connector is able to be bundled by modern JS toolchains
   3.a. Optionally, we can start including bundled distributions for those wishing to build browser/non-Node environment bots.

Comment (joshgummersall) [reply libraries/botbuilder-core/package.json:24]: Okay - I've rebased from `main` which now includes the `browserify`d `botframework-connector` package, so this should be resolved.

Comment (stevengum) [reply libraries/botbuilder-core/package.json:24]: Perfect, thanks!

**Review by stevengum — APPROVED** [review]
LGTM

**Pull request merged**
